content-signup/src/content-signup.js

```diff
@@ -0,0 +1,105 @@
+// Email template, paste Google Doc ID to display email format.
+// Make sure the Google Doc is shareable.
+const EMAIL_TEMPLATE_DOC_ID = '1j8MSwn6vK3Y_0uRzGV0Kok8AT5r1NbycoGimN0jYv68';
+
+// Sheet column names.
+const TIMESTAMP = 'timestamp';
+const NAME = 'myName';
+const EMAIL = 'emailAddress';
+const TOPICS = 'onTheFollowingTopics';
+const IN_MAILING_LIST = 'iWouldAlsoLikeToKeepReceivingInformationInTheFuture';
+const EMAIL_SENT_DATE = 'emailSentDate';
+
+// Links to send based on topics selected in form by user.
+const TOPIC_URLS = {
+  'Nutrition': 'https://docs.google.com/forms/d/e/1FAIpQLSch2B-ac_kS9ENNNlGeJK3Lba6Cs-UvjX5i4vNHjBnZYBv0kw/viewform',
+  'Reprogramming Habits': 'https://docs.google.com/forms/d/e/1FAIpQLSch2B-ac_kS9ENNNlGeJK3Lba6Cs-UvjX5i4vNHjBnZYBv0kw/viewform',
+  'Urban Food': 'https://docs.google.com/forms/d/e/1FAIpQLSch2B-ac_kS9ENNNlGeJK3Lba6Cs-UvjX5i4vNHjBnZYBv0kw/viewform',
+  'Water Design': 'https://docs.google.com/forms/d/e/1FAIpQLSch2B-ac_kS9ENNNlGeJK3Lba6Cs-UvjX5i4vNHjBnZYBv0kw/viewform',
+};
+
+// Get the active spreadsheet.
+var ss = SpreadsheetApp.getActive();
+
+// This is only used to ask for Drive scope permissions, used by the
+//`docToHtml` function.
+DriveApp.getStorageUsed();
+
+// Must click this function from the top drop-down and click 'run' icon to
+// activate setup of Form to activate on submit.
+function setup() {
+  ScriptApp.newTrigger('onFormSubmit')
+    .forSpreadsheet(ss)
+    .onFormSubmit()
+    .create();
+}
+
+// Reviews each row to see who to send an email to.
+function onFormSubmit() {
+  let sheet = ss.getSheets()[0];
+  let rows = sheet.getDataRange().getValues();
```

Comment:
Loading all the rows for every form submit won't be a problem at first, but as the spreadsheet gets longer it will start making this code rather slow. Consider either just processing the row that was just added (the active range) or perhaps using a property to record the last row you've processed.

Reply:
Noted, I have will work on this on the next wave of changes.